microsetta_admin/metadata_util.py

```diff
@@ -0,0 +1,141 @@
+from microsetta_admin._api import APIRequest
+from collections import defaultdict
+
+
+# TODO: If this needs to be able to handle every barcode in the system at once
+#  we will need to refactor to iterate over barcodes and append to some file.
+def retrieve_metadata(sample_barcodes):
```

Comment:
It would be beneficial to decompose this further... the method is rather complex at the moment